crates/sui-core/src/execution_scheduler/execution_scheduler_impl.rs

```diff
@@ -220,6 +321,18 @@ impl ExecutionScheduler {
         };
     }
 
+    async fn wait_barrier(&self, key: (ObjectID, SequenceNumber), count: u32) {
```

Comment:
Can you explain how this deals with races between checkpoint builder and checkpoint executor?